02/app.js

```diff
@@ -1 +1,7 @@
-console.log('DOM');
\ No newline at end of file
+const linksList = document.querySelectorAll("[data-url");
+linksList.forEach(function (element) {
+  if (element.hasAttribute("data-url")) {
```

Comment:
Jeśli wyszukujesz elementy pod atrybucie tj. `[data-url]` to nie ma sensu sprawdzać tego ponownie - strata czasu :)